src/middlewares/createInsightsMiddleware.ts

```diff
@@ -88,6 +88,13 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = props => {
     return {
       onStateChange() {},
       subscribe() {
+        if (
+          typeof instantSearchInstance.client.addAlgoliaAgent === 'function'
+        ) {
+          instantSearchInstance.client.addAlgoliaAgent(`insights-middleware`);
+        }
```

Comment:
```suggestion
```

this doesn't effect how the search behaves, so I don't think it should be added

Reply:
If we add it to insights call, we can know

> What percentage of insights calls are done by the insights middleware?

If we add it to search call, we can know

> What percentage of InstantSearch users are using the insights middleware?

That's why I added to both.

Reply:
Can't we rather add a similar `addAlgoliaAgent` method to the Insights client?

Reply:
that's already done @francoischalifour, check the lower lines :)

> What percentage of InstantSearch users are using the insights middleware?

I feel like that should rather be part of the metadata middleware

Reply:
> this doesn't effect how the search behaves

> I feel like that should rather be part of the metadata middleware

+ Without it, I think we could probably combine data to figure this out.

So let's remove the user agent from the search call for now.
https://github.com/algolia/instantsearch.js/pull/4748/commits/ec4ecc006677827d7efd00f9cb64d19ef478b43e